x-pack/plugins/observability_solution/inventory/public/components/entity_icon/index.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { AGENT_NAME, CLOUD_PROVIDER } from '@kbn/observability-shared-plugin/common';
+import { type CloudProvider, CloudProviderIcon, AgentIcon } from '@kbn/custom-icons';
+import { EuiFlexGroup, EuiFlexItem, EuiIcon } from '@elastic/eui';
+import type { AgentName } from '@kbn/elastic-agent-utils';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { ENTITY_TYPE } from '../../../common/es_fields/entities';
+import type { Entity } from '../../../common/entities';
+
+interface EntityIconProps {
+  entity: Entity;
+}
+
+type NotNullableCloudProvider = Exclude<CloudProvider, null>;
+
+const getSingleValue = <T,>(value?: T | T[] | null): T | undefined => {
+  return value == null ? undefined : Array.isArray(value) ? value[0] : value;
+};
+
+export function EntityIcon({ entity }: EntityIconProps) {
+  const entityType = entity[ENTITY_TYPE];
+  const defaultIconSize = euiThemeVars.euiSizeL;
+
+  switch (entityType) {
+    case 'host':
+    case 'container': {
+      const cloudProvider = getSingleValue(
+        entity[CLOUD_PROVIDER] as NotNullableCloudProvider | NotNullableCloudProvider[]
```

Comment:
Q: why is the cast `NotNullableCloudProvider` needed?

Reply:
I just added it.

`CloudProvider` can be `null`, but `CloudProviderIcon`'s `title` prop doesn't accept `null` values. To maintain consistent typings and avoid using `title={cloudProvider || undefined}`, I defined this specific type.


<img width="245" alt="Screenshot 2024-10-02 at 15 05 37" src="https://github.com/user-attachments/assets/c39db5a6-e00c-4ee0-8b92-20965d3ba72f">


I'm fine with using `title={cloudProvider || undefined}` if you think it's a better approach.